src/Core/src/ImageSources/Android/Glide/RequestBuilderExtensions.cs

```diff
@@ -3,31 +3,57 @@
 using System.Threading.Tasks;
 using Android.Content;
 using Android.Graphics.Drawables;
+using Android.Runtime;
 using Bumptech.Glide;
+using Bumptech.Glide.Load;
+using Bumptech.Glide.Load.Engine;
+using Bumptech.Glide.Request.Target;
+using Java.Lang;
 
 namespace Microsoft.Maui.BumptechGlide
 {
 	public static class RequestBuilderExtensions
 	{
-		public static async Task<Drawable?> SubmitAsync(this RequestBuilder requestBuilder, CancellationToken cancellationToken = default)
-		{
-			var target = requestBuilder.Submit();
-
-			return await target.AsTask<Drawable>(cancellationToken);
-		}
-
 		public static Task<ImageSourceServiceResult?> SubmitAsync(this RequestBuilder requestBuilder, Context context, CancellationToken cancellationToken = default) =>
 			requestBuilder.SubmitAsync(Glide.With(context), cancellationToken);
 
 		public static async Task<ImageSourceServiceResult?> SubmitAsync(this RequestBuilder requestBuilder, RequestManager requestManager, CancellationToken cancellationToken = default)
 		{
+			var callback = new RequestListener();
```

Comment:
The only thing tricky here is I don't know if `callback` will be kept alive the lifetime of this method.

You might wrap the whole thing in a `try-finally` block like:
```csharp
var callback = new RequestListener();
try
{
    // the other code here
}
finally
{
    GC.KeepAlive(callback);
}
```

`GC.KeepAlive()` doesn't do anything, it just tells the GC you're using it up to the point it's called.

We could look at the `.dll` when the build finishes. It might be worth knowing if async/await does the right thing.

Reply:
It seems like the generated async-state machine wouldn't hold onto `callback`:

![image](https://user-images.githubusercontent.com/840039/144882533-4a70816d-83ed-475c-a81e-5772777409ce.png)

![image](https://user-images.githubusercontent.com/840039/144882611-2b812a1e-624b-488e-89e7-1d6e29e67201.png)

Reply:
Nice catch